datacontract/model/data_contract_specification.py

```diff
@@ -31,6 +31,10 @@ class Server(pyd.BaseModel):
     token: str = None  # Use ENV variable
     dataProductId: str = None
     outputPortId: str = None
+    trusted_connection: str = None
+    trust_server_certificate: str = None
+    encrypt_connection: str = None
+    driver: str = None # Use ENV variable
```

Comment:
do you need those fields? could we get by without them?

Reply:
The default provided values are not entirely suitable for a large majority
of connections. Users that are using Kerberos or active directory, for
example require a trusted connection and will not provide a username or
password.

Reply:
> do you need those fields? could we get by without them?

I am open to alternatives if you'd rather have them be defined as environmental variables, for example

Reply:
I'd go for environment variables and sensible defaults, as they probably would not be part of the datacontract.yaml

Reply:
Hm, good question. I am no export of sqlserver, so I don't really know which way is better.

For me, env variables should be used for secrets or things that differ from user to user.

If, however, you need those fields in order to know how to connect to the data, and it can only be done knowing those infos, then we should add them to the spec. 

Does this help? What would you do with this guideline?

PS: I just noticed that the three fields are of type boolean, but represented as strings in the pydantic data contract spec model.

Reply:
@simonharrer I wanted to match the current code style as much as possible and use string representation as that seemed to be the default.

The only one absolutely required without any reasonable defaults is MSSQL driver. The others can be made as env variables